data-prepper-core/src/test/java/com/amazon/dataprepper/server/DataPrepperServerTest.java

```diff
@@ -103,11 +112,12 @@ public void stopServer() {
 
     @Test
     public void testGetGlobalMetrics() throws IOException, InterruptedException, URISyntaxException {
-        setupDataPrepper();
+        when(dataPrepperConfiguration.getServerPort()).thenReturn(port);
```

Comment:
It would be possible, two considerations I had.

1. Not all tests call this method. This way I can use the default behavior that if I do not use a mocked method the test will fail by default.
2. In the next pull request for adding dependency inject, DataPrepperServer will be refactors and this will no longer be required.